pkg/server/util/regentryutil/fetch.go

```diff
@@ -25,8 +33,14 @@ func newRegistrationEntryFetcher(dataStore datastore.DataStore) *registrationEnt
 	}
 }
 
-func (f *registrationEntryFetcher) Fetch(ctx context.Context, id string) ([]*common.RegistrationEntry, error) {
-	entries, err := f.fetch(ctx, id, make(map[string]bool))
+func (f *registrationEntryFetcher) Fetch(ctx context.Context, cache *FetchRegistrationEntriesCache, id string) ([]*common.RegistrationEntry, error) {
+	var entries []*common.RegistrationEntry
+	var err error
+	if cache != nil {
+		entries, err = f.fetchWithCache(ctx, id, cache, make(map[string]bool))
```

Comment:
Can extract out initialization of `visited` map since it's common to both branches